build.fsx

```diff
@@ -0,0 +1,120 @@
+        |> Paket.Dependencies 
+        |> fun d -> d.FindReferencesFor(Paket.Domain.MainGroup, packageName)
+
+module AppVeyor = 
+    let BuildNumber = environVarOrNone "APPVEYOR_BUILD_NUMBER"
+
+module Xml = 
+    open System.Xml.Linq
+    let load x = XDocument.Load(x:string)
+    let xns x = (x:XDocument).Root.Name.Namespace
+    let xn2 n xn = (xn:XNamespace).GetName(n)
+    let descendants n x = (x:XDocument).Descendants(n)
+    let value x = (x:XElement).Value
+
+module MsBuildProject = 
+    open System
+    open Xml
+    let tryAssembly project = 
+        let xdoc = Xml.load project 
+        let xmlns = xdoc |> Xml.xns
+        xdoc |> Xml.descendants (xmlns |> Xml.xn2 "AssemblyName") |> Seq.map Xml.value |> Seq.tryHead
+        |> Option.bind (fun assemblyName -> 
+            !! (directory project </> "bin" </> sprintf "**/%s.dll" assemblyName)
+            |> Seq.tryHead)
+
+module Analysis = 
+    let findAssembliesReferencing packageName = 
+        Paket.findReferencesFor packageName
+        |> Seq.collect (MsBuildProject.tryAssembly >> Option.toList)
+
+open AppVeyor
+open Fake.Testing.NUnit3
+
+module ReleaseNotes = 
+    let TickSpec = ReleaseNotesHelper.LoadReleaseNotes (__SOURCE_DIRECTORY__ </> "RELEASE_NOTES.md")
+
+module Build = 
+    let rootDir = __SOURCE_DIRECTORY__
+    let nuget = rootDir </> "packed_nugets"
+    let setParams defaults =
+        { defaults with
+            Verbosity = Some MSBuildVerbosity.Normal
+            Targets = ["Rebuild"]
+            Properties =
+                [ "AllowedReferenceRelatedFileExtensions", ".pdb"
+                  "Optimize", "True"
+                  "DebugSymbols", "True"
+                  "Configuration", "Release" ]
+         }
+
+let Sln = "./TickSpec.sln"
+
+Target "Clean" <| fun _ -> DeleteDir Build.nuget
+    
+Target "AssemblyInfo" <| fun _ ->
+    !! ("TickSpec" </> "AssemblyInfo.fs")
+    |> Seq.iter(fun asmInfo ->
+        let fileVersion =
+            BuildNumber |> Option.defaultValue "0" 
+            |> sprintf "%s.%s" ReleaseNotes.TickSpec.AssemblyVersion
+        [ Attribute.Version fileVersion
+          Attribute.FileVersion fileVersion ]
+        |> AssemblyInfoFile.UpdateAttributes asmInfo)
+
+Target "Build" <| fun _ -> build Build.setParams Sln
+
+Target "Test" <| fun _ ->
+    Analysis.findAssembliesReferencing "nunit"
+    |> NUnit3 (fun p -> 
+        { p with 
+            ToolPath = Build.rootDir </> "packages/build/NUnit.ConsoleRunner/tools/nunit3-console.exe" })
+    
+    Analysis.findAssembliesReferencing "xunit"
+    |> XUnit2.xUnit2 (fun p -> 
+        { p with 
+            ToolPath = Build.rootDir </> "packages/build/xunit.runner.console/tools/net452/xunit.console.exe"
+            ForceAppVeyor = AppVeyor.BuildNumber |> Option.isSome })
+
+Target "Nuget" <| fun _ -> 
+    Paket.Pack (fun p ->
+        { p with
+            BuildPlatform = "AnyCPU"
+            MinimumFromLockFile = true
```

Comment:
Can the paket publish template get deleted ? have we figured out whether the default meta for the package remains as it was before (and e.g. has not grown?)

Reply:
The dependencies are not really ised since it was hard code into the tickspec.paket.template (I think you could separate example from rhe lib by using another group)

Reply:
My only concern is to avoid duplication and/or confusion between the `paket.template` and this block here, i.e. the ideal is if this block is minimal in size (though one wouldnt want to publish any other way than this, the emphasis should be on removing as much as possible from the template). This is not a blocker would be nice to get done for once and for all

Reply:
done